File: lung_assignment.py

import pandas as pd
from scipy.stats import chi2_contingency
import statsmodels.formula.api as smf
from sklearn.metrics import roc_auc_score

# Load dataset
lung = pd.read_csv("lung_disease.csv")

# Crosstab
print(pd.crosstab(lung['Smoking'], lung['LungDisease']))

# Chi-square test
ct = pd.crosstab(lung['Smoking'], lung['LungDisease'])
chi = chi2_contingency(ct)
print(chi)

# Logistic Regression
lung['LD'] = lung['LungDisease'].map({'Yes':1,'No':0})

model = smf.logit(
    'LD ~ C(Smoking) + Age + C(Pollution) + C(Income)',
    data=lung
).fit()

print(model.summary())

# AUC
pred = model.predict(lung)
auc = roc_auc_score(lung['LD'], pred)
print("AUC =", auc)

File: lung_assignment.R

# Load Dataset
lung <- read.csv("lung_disease.csv")

# Crosstab
table(lung$Smoking, lung$LungDisease)

# Chi-square test
chisq.test(table(lung$Smoking, lung$LungDisease))

# Logistic Regression
lung$LD <- ifelse(lung$LungDisease=="Yes",1,0)

model <- glm(
LD ~ Smoking + Age + Pollution + Income,
data=lung,
family=binomial
)

summary(model)

# ROC & AUC
library(pROC)

pred <- predict(model, type="response")
roc_obj <- roc(lung$LD, pred)
auc(roc_obj)
